### PR TITLE
Feature/location query

### PR DESCRIPTION
**Pull request by akolliy1**
#### What does this PR do?
- use HTML Location
- have Geo IP as fallback https://ipapi.co/json/
- check if Geo IP as fallback would be allowed under the GDPR/DSGVO

### CODE REVIEW

File: frontend/src/context/useCurrentLocation.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  const [state, dispatch] = useReducer(loggerReducer, data)
+
+
+  useEffect(() => {
+    setData(state)
+  }, [state, setData])
+
+  const { selectedLocation, coordinates } = state
+
+  const addCoordinates = coordinates => {
```
Comment (vellip) [inline frontend/src/context/useCurrentLocation.js:54]: Should it not be called `setCoordinates`?

File: frontend/src/context/useCurrentLocation.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      navigator.geolocation.getCurrentPosition(
+        (position) => {
+          const lattitude = position.coords.latitude;
+          const longitude = position.coords.longitude;
+          const coordinates = [longitude, lattitude];
+          // TODO: loader status: setIsGettingLocation(false);
+          addCoordinates(coordinates);
+        },
+        () =>
+          setIsGettingLocation(false)
```
Comment (vellip) [inline frontend/src/context/useCurrentLocation.js:73]: Should be commented out, no?

File: frontend/src/context/useCurrentLocation.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      );
+    } else {
+      // Browser doesn't support Geolocation
+      axios.get('https://ipapi.co/json/')
+      .then((response) => {
+        const { latitude, longitude } = response;
+        const coordinates = [longitude, latitude];
+        addCoordinates(coordinates);
```
Comment (vellip) [inline frontend/src/context/useCurrentLocation.js:81]: should set LoaderStatus false here as well

File: frontend/src/context/useCurrentLocation.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  };
+
+  const addGoogleLocation = (suggestion, coordinates) => {
+    dispatch({
+      type: 'SET_SELECTED_LOCATION',
+      payload: {suggestion, coordinates}
+    })
+  }
+
+  return { selectedLocation, coordinates, addCurrentLocation, addGoogleLocation }
```
Comment (vellip) [inline frontend/src/context/useCurrentLocation.js:97]: Should they be called `add` or `set`?

File: frontend/src/context/useCurrentLocation.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+};
+
+const loggerReducer = logger(reducer);
+
+const useCurrentLocation = () => {
+  //  TODO: loader status: 
+  // const [isGettingLocation, setIsGettingLocation] = useState(false);
+  const [data, setData] = useLocalStorage('locationData', initialState)
```
Comment (vellip) [inline frontend/src/context/useCurrentLocation.js:44]: I think there is no need for this to be stored in LocalStorage, you can just use `useState`

File: frontend/src/context/useCurrentLocation.js (hunk 4 above)
Comment (akolliy1) [reply frontend/src/context/useCurrentLocation.js:97]: I prefered set.

File: frontend/src/context/useCurrentLocation.js (hunk 5 above)
Comment (akolliy1) [reply frontend/src/context/useCurrentLocation.js:44]: Well, it depends. we can use `useState` and later if it occurred we need persistence. we can refactor the part.

File: frontend/src/context/useCurrentLocation.js [hunk 6]
```diff
@@ -67,18 +67,19 @@ const useCurrentLocation = () => {
           const longitude = position.coords.longitude;
           const coordinates = [longitude, lattitude];
           // TODO: loader status: setIsGettingLocation(false);
-          addCoordinates(coordinates);
+          setSelectedLocation(coordinates);
         },
-        () =>
-          setIsGettingLocation(false)
+        () => // TODO: loader status: setIsGettingLocation(false)
```
Comment (vellip) [inline frontend/src/context/useCurrentLocation.js:72]: Does that not throw an error?

**Pull request merged**